packages/opentelemetry-sdk-trace-web/src/utils.ts

```diff
@@ -89,28 +89,32 @@ export function addSpanNetworkEvent(
 }
 
 /**
- * Helper function for adding network events
+ * Helper function for adding network events and content length attributes
  * @param span
  * @param resource
+ * @param ignoreNetworkEvents
  */
 export function addSpanNetworkEvents(
   span: api.Span,
-  resource: PerformanceEntries
+  resource: PerformanceEntries,
+  ignoreNetworkEvents = false
```

Comment:
hmm, that's really unfortunate.

I'd really like to avoid carrying this forward to 2.0 - @arriIsHere would you mind opening a follow-up PR towards the `next` branch later to implement option 1 "new helper method" from [this comment](https://github.com/open-telemetry/opentelemetry-js/pull/5230#issuecomment-2517876297)) there? It adds one more function but we'll be able to get rid of the body-size side-effect in this function, which I think cleans it up a bit.

I plans to move `@opentelemetry/instrumentation-xhr` and `@opentelemetry/instrumentation-fetch` into the same package as a follow up, which may even let us drop these functions from the public API of `@opentelemetry/sdk-trace-web` since they're not really SDK functions. That'd also go a long way of discouraging the anti-pattern of direct depending on the SDK for instrumentation.